CIME/XML/env_batch.py

```diff
@@ -964,7 +963,17 @@ def _submit_single_job(
 
             return
 
-        submitargs = case.get_value("BATCH_COMMAND_FLAGS", subgroup=job)
+        submitargs = case.get_value("BATCH_COMMAND_FLAGS", subgroup=job, resolved=False)
+
+        project = case.get_value("PROJECT", subgroup=job)
+
+        if not project:
+            # If there is no project then we need to remove the project flag
+            # slurm defines --account only on machines that require it, so this strip isn't required
+            if batch_system == "pbs" or batch_system == "cobalt":
```

Comment:
There's at least one cobalt variant "cobart_theta", maybe the match should be a pattern match rather than an exact match?

Reply:
cobalt_theta is only defined for one machine and that machine has PROJECT_REQUIRED=TRUE.

Reply:
Awesome.